sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -394,7 +394,9 @@ object GpuOrcScan {
               withResource(doubleMillis.add(half)) { doubleMillisPlusHalf =>
                 withResource(doubleMillisPlusHalf.floor()) { millis =>
                   withResource(getOverflowFlags(doubleMillis, millis)) { overflowFlags =>
-                    millis.copyWithBooleanColumnAsValidity(overflowFlags)
+                    withResource(Scalar.fromNull(millis.getType)) { NULL =>
+                      overflowFlags.ifElse(millis, NULL)
```

Comment:
In general we should avoid using keyword names such as `null` or `NULL`.  
```suggestion
                    withResource(Scalar.fromNull(millis.getType)) { nullVal =>
                      overflowFlags.ifElse(millis, nullVal)
```

Reply:
Done.